src/interfaces/api.ts

```diff
@@ -0,0 +1,10 @@
+export interface Product {
+  id: number;
+  title: string;
+}
+
+export interface Recipe {
+  id: number;
+  title: string;
+  items: string | number;
```

Comment:
Pretty sure it isn't a string or number. Its an array of a whole new type that you will need to declare
![image](https://user-images.githubusercontent.com/20260038/180863549-a836e16d-4739-4f0a-a277-60369300a690.png)

Reply:
how about a new interface
```ts
interface RecipeItem {
 ....
}

interface Recipe {
items: RecipeItem[]; // or Array<RecipeItem>
...
}
```

Reply:
A trick when specifing types/interfaces - if you declare a const with the values you expect, you can hover over the variable name to read what type it infers.
![type1](https://user-images.githubusercontent.com/20260038/180885568-05b5994c-c2f6-461a-bf6c-8648aa05f6b5.gif)

then you can take that type and declare an interface based on it, 
```ts
interface Recipe {
    id: number;
    title: string;
    items: {
        id: number;
        qty: number;
        unit: string;
    }[];
}
```

and usually if a property is an object I would declare a separate type for it to keep the structure flat.
```ts
interface RecipeItem {
        id: number;
        qty: number;
        unit: string;
    }

interface Recipe {
    id: number;
    title: string;
    items: RecipeItem[];
}
```